model/validation.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2016 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/golang/protobuf/proto"
+)
+
+const (
+	dns1123LabelMaxLength int    = 63
+	dns1123LabelFmt       string = "[a-z0-9]([-a-z0-9]*[a-z0-9])?"
+)
+
+var (
+	dns1123LabelRex = regexp.MustCompile("^" + dns1123LabelFmt + "$")
+	kindRegexp      = regexp.MustCompile("^[a-zA-Z][a-zA-Z0-9]*$")
+)
+
+// IsDNS1123Label tests for a string that conforms to the definition of a label in
+// DNS (RFC 1123).
+func IsDNS1123Label(value string) bool {
+	return len(value) <= dns1123LabelMaxLength && dns1123LabelRex.MatchString(value)
+}
+
+// Validate names in the config key
```

Comment:
nit: `// Validate confirms that the names in the ConfigKey are appropriately formatted.`